File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/TA/Mixed/NNTTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4objects.Db4o.Tests.Common.TA;
using Db4objects.Db4o.Tests.Common.TA.Mixed;

namespace Db4objects.Db4o.Tests.Common.TA.Mixed
{
	/// <exclude></exclude>
	public class NNTTestCase : ItemTestCaseBase
	{
		public static void Main(string[] args)
		{
			new NNTTestCase().RunAll();
		}

		/// <exception cref="System.Exception"></exception>
		protected override object CreateItem()
		{
			return new NNTItem(42);
		}

		/// <exception cref="System.Exception"></exception>
		protected override void AssertRetrievedItem(object obj)
		{
			NNTItem item = (NNTItem)obj;
			Assert.IsNotNull(item.ntItem);
			Assert.IsNotNull(item.ntItem.tItem);
			Assert.AreEqual(0, item.ntItem.tItem.value);
		}

		/// <exception cref="System.Exception"></exception>
		protected override void AssertItemValue(object obj)
		{
			NNTItem item = (NNTItem)obj;
			Assert.AreEqual(42, item.ntItem.tItem.Value());
		}

		/// <exception cref="System.Exception"></exception>
		public virtual void TestDeactivateDepth()
		{
			NNTItem item = (NNTItem)RetrieveOnlyInstance();
			NTItem ntItem = item.ntItem;
			TItem tItem = ntItem.tItem;
			tItem.Value();
			// item.ntItem.tItem.value
			Assert.IsNotNull(ntItem.tItem);
			Db().Deactivate(item, 2);
			// FIXME: failure 
			//		Assert.isNull(ntItem.tItem);
			Db().Activate(item, 42);
			Db().Deactivate(item, 3);
			Assert.IsNull(ntItem.tItem);
		}
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/common/ta/mixed/NNTTestCase.java
/* Copyright (C) 2007  Versant Inc.  http://www.db4o.com */

package com.db4o.db4ounit.common.ta.mixed;

import com.db4o.db4ounit.common.ta.*;

import db4ounit.*;

/**
 * @exclude
 */
public class NNTTestCase extends ItemTestCaseBase {
	
	public static void main(String[] args) {
		new NNTTestCase().runAll();
	}
	
	protected Object createItem() throws Exception {
		return new NNTItem(42);
	}

	protected void assertRetrievedItem(Object obj) throws Exception {
		NNTItem item = (NNTItem) obj;
		Assert.isNotNull(item.ntItem);
		Assert.isNotNull(item.ntItem.tItem);
		Assert.areEqual(0, item.ntItem.tItem.value);
	}
		
	protected void assertItemValue(Object obj) throws Exception {
		NNTItem item = (NNTItem) obj;
		Assert.areEqual(42, item.ntItem.tItem.value());
	}

	public void testDeactivateDepth() throws Exception {
		NNTItem item = (NNTItem) retrieveOnlyInstance();
		NTItem ntItem = item.ntItem;
		TItem tItem = ntItem.tItem;
		tItem.value();
		// item.ntItem.tItem.value
		Assert.isNotNull(ntItem.tItem);
		db().deactivate(item, 2);
		// FIXME: failure 
//		Assert.isNull(ntItem.tItem);
		
		db().activate(item, 42);
		db().deactivate(item, 3);
		Assert.isNull(ntItem.tItem);
	}
}
